apps/api/routes.py

```diff
@@ -194,6 +194,28 @@ def save_lab_answers(lab_inst_id):
 
     return {"status": "ok", "result": "Answers saved successfully"}, 200
 
+@blueprint.route('/lab_answers/correct/<lab_inst_id>', methods=["POST"])
+def correct_lab_answer(lab_inst_id):
+    if current_user.category == "user":
+        return {}, 404
+
+    lab_inst = LabInstances.query.get(lab_inst_id)
+    if not lab_inst:
+        return {"status": "fail", "result": "Lab instance not found"}, 404
+
+    lab_answers = LabAnswers.query.filter_by(lab_id=lab_inst_id, user_id=current_user.id).first()
+    if not lab_answers:
+        return {"status": "fail", "result": "Lab answers not found"}, 404
```

Comment:
@silvasilas99 on line 202 you already have the lab_instance object, so you dont need to do another query to the DB to validate the ownership as you are doing on line 206. You can just do a `if lab_inst.user_id != current_user.id`

However, I wonder what is the actual objective here. From the log message generated on line 208, you say lab answer not found, which does not make much sense with the validation being performed. Were you aiming to validate if the user has permission to change this lab answer?